api/incubator/src/main/java/io/opentelemetry/api/incubator/common/ArrayBackedExtendedAttributesBuilder.java

```diff
@@ -36,11 +47,126 @@ public <T> ExtendedAttributesBuilder put(ExtendedAttributeKey<T> key, T value) {
     if (key == null || key.getKey().isEmpty() || value == null) {
       return this;
     }
+    if (key.getType() == ExtendedAttributeType.VALUE && value instanceof Value) {
+      putValue(key, (Value<?>) value);
+      return this;
+    }
     data.add(key);
     data.add(value);
     return this;
   }
 
+  private void putValue(ExtendedAttributeKey<?> key, Value<?> valueObj) {
+    // Convert VALUE type to narrower type when possible
+    String keyName = key.getKey();
+    switch (valueObj.getType()) {
+      case STRING:
+        data.add(stringKey(keyName));
+        data.add(valueObj.getValue());
```

Comment:
Can use a pattern like: `put(stringKey(keyName), ((Value<String>) valueObj).getValue())` to save a few lines. 

Is there a way to be able to store the value key without having to convert it to a string / long / double / etc key first? I.e. something like:
- Always store the value key
- When fetching for a given key, say of type string, check:
  - If there is a matching key with type string
  - OR if there is a matching key of type value and with a value of type string

Reply:
> Can use a pattern like: `put(stringKey(keyName), ((Value<String>) valueObj).getValue())` to save a few lines.

575535d4ce30633200aee8b0a66e0455a3975738

Reply:
Ok, but there's obviously a steep perf penalty for making a copy of the attribute key every time. Maybe that's fine as it provides a disincentive for using `VALUE` where not necessary. Could also solve with an optimization in `ExtendedAttributeKey`, where we precompute and/or cache the primitive attribute key equivalent when the type is `VALUE`. Then, this code could leverage the preocmputed / cached value.